js/client.js

```diff
@@ -843,10 +843,27 @@
 						if (data === 'nonexistent' && Config.server.id && roomid.slice(0, 7) === 'battle-') {
```

Comment:
Also, if you're going to do this, don't do `unjoinRoom` first it – i.e. don't close and reopen the room.